core/src/main/java/com/scalar/db/transaction/consensuscommit/RecoveryExecutor.java

```diff
@@ -272,4 +314,10 @@ public Result(
       this.recoveryFuture = recoveryFuture;
     }
   }
+
+  public enum RecoveryType {
+    RETURN_LATEST_RESULT_AND_RECOVER,
+    RETURN_COMMITTED_RESULT_AND_RECOVER,
+    RETURN_COMMITTED_RESULT_AND_NOT_RECOVER
+  }
```

Comment:
Introduce `RecoveryType` in `RecoveryExecutor`.

`RETURN_LATEST_RESULT_AND_RECOVER`: This is the previous behavior. Return the latest record with the coordinator check, then start lazy recovery.
`RETURN_COMMITTED_RESULT_AND_RECOVER`: Return a record created from the before image without the coordinator check, then start lazy recovery.
`RETURN_COMMITTED_RESULT_AND_NOT_RECOVER`: Return a record created from the before image without the coordinator check. This type does not start lazy recovery.